### PR TITLE
Automated cherry pick of #114464: Set permissions on volume before publishing update

### PR DESCRIPTION
**Pull request by furkatgofurov7**
Cherry pick of #114464 on release-1.26.

#114464: Set permissions on volume before publishing update

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

```release-note
Fixed file permission issues that happened during update of Secret/ConfigMap/projected volume when fsGroup is used. The problem caused a race condition where application gets intermittent permission denied error when reading files that were just updated, before the correct permissions were applied.
```

### CODE REVIEW

**Comment by furkatgofurov7** [comment]
@thockin hi, can you please take a look at this cherry-pick when you have time? Thanks!

**Review by thockin — COMMENTED** [review]
From a tech POV this is LGTM.  I think it's a real issue, but relatively low severity, so I would not endorse backporting beyond 1.26

**Comment by furkatgofurov7** [comment]
> From a tech POV this is LGTM. I think it's a real issue, but relatively low severity, so I would not endorse backporting beyond 1.26

@thockin Thanks for the input, this was the real issue in fact, and was escalated quickly on our side. I would really appreciate if we could get this merged so that it could make it to the next patch release.  Is there anything else I could do to get the green light (labels) from you?  Also pinging release managers here, PTAL.  

/cc kubernetes/release-managers

**Comment by xmudrii** [comment]
We're past the cherry-pick deadline. We'll get back to reviewing cherry-picks after cutting January patch releases.

**Comment by furkatgofurov7** [comment]
Hey folks, can we consider taking this cherry pick in to the upcoming patch release? 

/cc @xmudrii kubernetes/release-managers

**Comment by furkatgofurov7** [comment]
cc @tsaarni for possible follow-ups since we discussed it offline.

**Comment by tsaarni** [comment]
I've rebased this cherry pick request on top of changes introduced by #117575. The PR solves bug #114461 which is still an issue for us. Hoping for review and inclusion of this backport for [release-1.26](https://github.com/kubernetes/kubernetes/tree/release-1.26).

**Comment by furkatgofurov7** [comment]
> From a tech POV this is LGTM. I think it's a real issue, but relatively low severity, so I would not endorse backporting beyond 1.26

@thockin hi 👋🏼!

For cherry pick approval, looks like this needs lgtm/approve labels first, so reaching out again here and see the possibility of moving forward with this PR, as I understand there seems to be no concerns of backporting it to 1.26 from your side?

**Review by thockin — COMMENTED** [review]
/lgtm
/approve

**Review by xmudrii — APPROVED** [review]
For RelEng:
/lgtm
/approve

**Comment by xmudrii** [comment]
/release-note-edit
```release-note
Fixed file permission issues that happened during update of Secret/ConfigMap/projected volume when fsGroup is used. The problem caused a race condition where application gets intermittent permission denied error when reading files that were just updated, before the correct permissions were applied.
```

**Pull request merged**
